VectorPairHMMUnitTest fails if the machine supports AVX but the library isn't built

`VectorPairHMMUnitTest` checks `isAVXSupported()` and skips tests if it's not.  However, if AVX is supported on the machine, but the library isn't available it fails with 

```
Caused by: java.lang.IllegalArgumentException: Resource not found relative to class org.broadinstitute.hellbender.utils.pairhmm.VectorLoglessPairHMM: /lib/libVectorLoglessPairHMM.so
	at org.broadinstitute.hellbender.utils.io.Resource.getResourceContentsAsStream(Resource.java:60)
	at org.broadinstitute.hellbender.utils.io.IOUtils.writeResource(IOUtils.java:234)
	at org.broadinstitute.hellbender.utils.io.IOUtils.writeTempResource(IOUtils.java:222)
	at org.broadinstitute.hellbender.utils.pairhmm.VectorLoglessPairHMM.<init>(VectorLoglessPairHMM.java:106)
	at org.broadinstitute.hellbender.utils.pairhmm.VectorPairHMMUnitTest.<init>(VectorPairHMMUnitTest.java:22)
	... 46 more
```

This should be fixed so that the tests are skipped instead.

--- a/src/main/java/org/broadinstitute/hellbender/exceptions/UserException.java
+++ b/src/main/java/org/broadinstitute/hellbender/exceptions/UserException.java
@@ -388,5 +388,9 @@ public class UserException extends RuntimeException {
         public HardwareFeatureException(String message) {
             super(message);
         }
+
+        public HardwareFeatureException(String message, Exception e){
+            super(message, e);
+        }
     }
 }

--- a/src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java
+++ b/src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java
@@ -2,15 +2,15 @@ package org.broadinstitute.hellbender.utils.pairhmm;
 
 import org.apache.log4j.Logger;
 import org.broadinstitute.hellbender.exceptions.UserException;
-import org.broadinstitute.hellbender.utils.genotyper.ReadLikelihoods;
 import org.broadinstitute.hellbender.utils.genotyper.LikelihoodMatrix;
 import org.broadinstitute.hellbender.utils.haplotype.Haplotype;
-import org.broadinstitute.hellbender.utils.read.GATKRead;
-import org.broadinstitute.hellbender.utils.read.ReadUtils;
 import org.broadinstitute.hellbender.utils.io.IOUtils;
 import org.broadinstitute.hellbender.utils.io.Resource;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.ReadUtils;
 
-import java.io.*;
+import java.io.File;
+import java.io.IOException;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
@@ -22,6 +22,7 @@ public final class VectorLoglessPairHMM extends LoglessPairHMM {
 
     final static Logger logger = Logger.getLogger(VectorLoglessPairHMM.class);
     final static Boolean runningOnMac = System.getProperty("os.name", "unknown").toLowerCase().startsWith("mac");
+    private static final String AVX_NATIVE_CODE_PATH_IN_JAR = "/lib/libVectorLoglessPairHMM";
     long threadLocalSetupTimeDiff = 0;
     long pairHMMSetupTime = 0;
     static Boolean isVectorLoglessPairHMMLibraryLoaded = false;
@@ -101,12 +102,17 @@ public final class VectorLoglessPairHMM extends LoglessPairHMM {
                     logger.info("libVectorLoglessPairHMM found in JVM library path");
                 } catch (UnsatisfiedLinkError e) {
                     //Could not load from Java's library path - try unpacking from jar
-                    logger.debug("libVectorLoglessPairHMM not found in JVM library path - trying to unpack from GATK jar file");
-                    String path = "/lib/libVectorLoglessPairHMM" + (runningOnMac ? ".dylib" : ".so");
-                    File temp = IOUtils.writeTempResource(new Resource(path, VectorLoglessPairHMM.class));
-                    temp.deleteOnExit();
-                    System.load(temp.getAbsolutePath());
-                    logger.info("libVectorLoglessPairHMM unpacked successfully from GATK jar file");
+                    try {
+                        logger.debug("libVectorLoglessPairHMM not found in JVM library path - trying to unpack from GATK jar file");
+                        String path = AVX_NATIVE_CODE_PATH_IN_JAR + (runningOnMac ? ".dylib" : ".so");
+                        File temp = IOUtils.writeTempResource(new Resource(path, VectorLoglessPairHMM.class));
+                        temp.deleteOnExit();
+                        System.load(temp.getAbsolutePath());
+                        logger.info("libVectorLoglessPairHMM unpacked successfully from GATK jar file");
+                    } catch (final Exception furtherException){
+                        throw new UserException.HardwareFeatureException("Machine supports AVX, but failed to load the AVX PairHMM library from the classpath.  " +
+                                "Check that your jar contains native code for your platform or choose a java HMM.", furtherException);
+                    }
                 }
                 logger.info("Using vectorized implementation of PairHMM");
                 isVectorLoglessPairHMMLibraryLoaded = true;

--- a/src/test/java/org/broadinstitute/hellbender/utils/pairhmm/VectorPairHMMUnitTest.java
+++ b/src/test/java/org/broadinstitute/hellbender/utils/pairhmm/VectorPairHMMUnitTest.java
@@ -8,6 +8,7 @@ import org.broadinstitute.hellbender.utils.read.ReadUtils;
 import org.broadinstitute.hellbender.utils.test.BaseTest;
 import org.broadinstitute.hellbender.utils.text.parsers.BasicInputParser;
 import org.testng.Assert;
+import org.testng.SkipException;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
@@ -19,23 +20,28 @@ import java.util.*;
 
 public final class VectorPairHMMUnitTest extends BaseTest {
     private final static boolean DEBUG = false;
-    final N2MemoryPairHMM avxPairHMM = new VectorLoglessPairHMM();
 
     private static final String publicTestDirRelative = "src/test/resources/";
     public static final String publicTestDir = new File(gatkDirectory, publicTestDirRelative).getAbsolutePath() + "/";
     public static final String pairHMMTestData = publicTestDir + "pairhmm-testdata.txt";
 
-    @BeforeClass
+   @BeforeClass
     public void initialize() {
-        avxPairHMM.doNotUseTristateCorrection();
-
         if (!VectorLoglessPairHMM.isAVXSupported()) {
-          throw new org.testng.SkipException("AVX is not supported on this system.");
+          throw new SkipException("AVX is not supported on this system.");
+        } else {
+            try {
+               new VectorLoglessPairHMM();
+            } catch (final Exception e){
+                throw new SkipException("AVX library not available");
+            }
         }
     }
 
     private List<N2MemoryPairHMM> getHMMs() {
-        return Arrays.asList(avxPairHMM);
+        final N2MemoryPairHMM avxPairHMM = new VectorLoglessPairHMM();
+        avxPairHMM.doNotUseTristateCorrection();
+        return Collections.singletonList(avxPairHMM);
     }
 
     // --------------------------------------------------------------------------------

